### PR TITLE
Feature/main configuration

### PR DESCRIPTION
**Pull request by bobapple**
This PR adds management of `icinga2.conf` and `constants.conf`

I'm not very happy about the `i2_const_*` variable. I would rather have a `i2_constants` dictionary where the user can overwrite certain values. To make this possible the user would be required to set the Ansible variable `DEFAULT_HASH_BEHAVIOUR` to `merge` which is [not recommended](http://docs.ansible.com/ansible/latest/reference_appendices/config.html#default-hash-behaviour) by Ansbile.

Feedback is very welcome.

### CODE REVIEW

File: vars/Debian.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+i2_user: nagios
```
Comment (aflatto) [inline vars/Debian.yml:2]: Excuse me ???

**Review by aflatto — CHANGES_REQUESTED** [review]
You can use with_dict to have the variables as a dictionary and not all defined in individual keys.

then you can iterate over the dictionary, check if the value is null or default using the "ternary" logic and thus create the template.

File: vars/Debian.yml (hunk 1 above)
Comment (mcktr) [reply vars/Debian.yml:2]: On Debian the `nagios` user is used for historical reasons, so this should be fine.

Comment (bobapple) [reply vars/Debian.yml:2]: yup, `nagios` is default on debian

**Comment by bobapple** [comment]
How can i override a single value in a dictionary when using `with_dict`? Lets say I have the following dictionary:

```yaml
i2_constants:
  plugindir: "{{ i2_lib_dir }}/nagios/plugins"
  manubulonplugindir: "{{ i2_lib_dir }}/nagios/plugins"
  plugincontribdir: "{{ i2_lib_dir }}/nagios/plugins"
  nodename: "{{ ansible_fqdn }}"
  zonename: "{{ ansible_fqdn }}"
```
How would I override only `i2_constants.nodename` in my playbook?

**Comment by bobapple** [comment]
Another way that comes to my mind is using the `combined` filter with two separate dicts. I'n not sure if this is actually a practicable way, perhaps it would just overcomplicate things

**Comment by aflatto** [comment]
i2_constants.nodename: "{{ (i2_constants.nodename) | ternary(i2_constants.nodename, SOME_VALUE) }}"
you can see the example here:
https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html

**Comment by bobapple** [comment]
The  problem is that, if I set a new value for `i2_constants.nodename`, the rest of the dict will be empty. Because Ansible does not merge hashes by default. That's why I was referring to `DEFAULT_HASH_BEHAVIOUR`

**Comment by RossBarnie** [comment]
I think in the long run the `combine` option would be more maintainable than adding a ternary every time we need a variable.

Changing the `DEFAULT_HASH_BEHAVIOUR` to `merge` will make the role compatible only with other roles/plays with the `merge` behaviour, as far as I'm aware.

**Comment by aflatto** [comment]
I agree with @RossBarnie , can you alter the Code to use the combine filter ?

**Comment by RossBarnie** [comment]
I've got a working solution on https://github.com/RossBarnie/ansible-icinga2/commit/cea9508957ed85a737cf59f526b090fbd60c39d7. We lose the comments from the template but otherwise much cleaner. Not sure about the variable naming but seemed the most accurate.

EDIT: Fixed double definition of `i2_constants`.

**Comment by RossBarnie** [comment]
I've added the backup of the original constants file as requested by @aflatto. https://github.com/Icinga/ansible-icinga2/compare/feature/main-configuration...RossBarnie:feature/main-configuration

**Pull request merged**